src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,356 @@
+        Event event = new Event(taskDetails[2], TaskType.EVENT, taskDetails[3]);
+        tasks.add(event);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //add todo from file
+    private static void addToDoFile(String[] taskDetails) {
+        ToDo toDo = new ToDo(taskDetails[2], TaskType.TODO);
+        tasks.add(toDo);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //save all list data
+    private static void saveData() {
+        try {
+            createFile();
+            FileWriter fw = new FileWriter(filepath, false);
+            for (Task task : tasks) {
+                task.saveToFile(fw);
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Cannot create file; reason: " + e.getMessage());
+        }
+    }
+
+    //create new dir if required
+    private static void createFile() throws IOException {
+        File storagefile = new File(filepath);
+        if (storagefile.exists()) {
+            return;
+        }
+        if (!storagefile.getParentFile().exists()) {
+            storagefile.getParentFile().mkdirs();
+        }
+        storagefile.createNewFile();
+    }
+
+    //add user input to list
+    public static void addList(String userInput) throws InvalidCommandException {
+        //first word of user's input
+        String userInputFirstWord = userInput.split(" ")[0];
+        //create task type based on user input
+        switch (userInputFirstWord) {
+        case "todo":
+            try {
+                addTodo(userInput);
+            } catch (EmptyTodoException e) {
+                displayEmptyTodoMessage();
+            }
+            break;
+        case "deadline":
+            addDeadline(userInput);
+            break;
+        case "event":
+            addEvent(userInput);
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
+        tasks.get(tasks.size() - 1).respondOnAdd();
+    }
+
+    //add event to list
+    private static void addEvent(String userInput) {
+        TaskType taskType;
+        String description;
+
+        //start and end index for description
+        int taskDescriptionStartIndex;
+        int taskDescriptionEndIndex;
+
+        //find start of task description
+        taskDescriptionStartIndex = userInput.indexOf(" ");
+        taskDescriptionEndIndex = userInput.indexOf("/at");
+        //assign task's information
+        description = userInput.substring(taskDescriptionStartIndex + 1, taskDescriptionEndIndex).trim();
+        String at = userInput.substring(taskDescriptionEndIndex + 3).trim();//+3 to exclude "/at"
+        taskType = TaskType.EVENT;
+        //create and assign information to new event
+        Event event = new Event(description, taskType, at);
+        tasks.add(event);
```

Comment:
Excessive comments (comments are not required when the code is self explanatory)